### ISSUE TITLE
possible or not to enable anonymous search using config file?

### ISSUE THREAD

**Issue by cmichon** [glauth/glauth#300]
glauth is a great tool, but I'm struggling trying to enable anonymous search using config file. Is it even possible?

**Comment by Fusion** [comment]
I have not touched the authentication area in a while,  but I believe that it is not currently possible.
There is a setting to allow querying the root record anonymously to learn about where things are before performing an authenticated query, but this is the extent of anonymous querying.

Is there a good use case for fully anonymous queries?

**Comment by cmichon** [comment]
yes there is a good use case, when the authentication is done via a DN that cannot be guess from a uid or a username. In that case, an anonymous search must be done first to retrieve the DN to enable a proper bind.

If it cannot be done with the current state, no pb. I even asked ChatGPT for a solution :-D  and it came up with a [accesscontrol] section with anonymous configurable bind and search. It did not work at all, that's when I suspected this is not working with the config file approach.

**Comment by candlerb** [comment]
The standard solution here is to create a "service account", with a fixed username/password, that the machine that wants to make a query uses. i.e.

1. Bind with service account credentials
2. Search for user record
3. Rebind with user DN and the user password
